src/WTG.Analyzers/Analyzers/Visibility/VisibilityAnalyzer.cs

```diff
@@ -65,6 +67,28 @@ static void Analyze(SyntaxNodeAnalysisContext context, FileDetailCache cache)
 			}
 		}
 
+		static bool DoesMethodHaveReturnValueOrOutKeyword(SyntaxNode node)
+		{
+			if (!node.IsKind(SyntaxKind.MethodDeclaration))
+			{
+				return false;
+			}
+
+			var methodNode = (MethodDeclarationSyntax)node;
+			if (methodNode.ReturnType is PredefinedTypeSyntax predefinedTypeSyntax
```

Comment:
Is there a Kind check we can perform on `methodNode.ReturnType` before turning to runtime type casts?

Reply:
I suppose the same as Kind check for `var methodNode = (MethodDeclarationSyntax)node;`, e.g.
```csharp
if (methodNode.ReturnType.IsKind(SyntaxKind.PredefinedType))
{
	var returnType = (PredefinedTypeSyntax)methodNode.ReturnType;
	if (!returnType.Keyword.IsKind(SyntaxKind.VoidKeyword))
	{
		return true;
	}
}
```